pkg/onchain/website/uploader.go

```diff
@@ -109,57 +120,47 @@ func upload(
 	operationBatch sendOperation.OperationBatch,
 ) ([]string, error) {
 	nbChunks := len(chunks)
-	operations := make([]string, nbChunks+1)
-
-	operationWithEventResponse, err := onchain.CallFunction(
-		client,
-		nickname,
-		addr,
-		"initializeWebsite",
-		convert.U64ToBytes(nbChunks),
-		sendOperation.OneMassa, // To be updated when readonly is working to estimate storage cost
-		sendOperation.DefaultSlotsDuration,
-		false,
-		operationBatch,
-		&signer.WalletPlugin{},
-	)
-	if err != nil {
-		return nil, fmt.Errorf("calling initializeWebsite at '%s': %w", addr, err)
-	}
+	operations := make([]string, nbChunks)
 
-	operations[0] = operationWithEventResponse.OperationResponse.OperationID
-
-	for index := 0; index < nbChunks; index++ {
-		chunkSize := len(chunks[index])
+	for chunkIndex := 0; chunkIndex < nbChunks; chunkIndex++ {
+		chunkSize := len(chunks[chunkIndex])
 		// Chunk ID encoding
-		params := convert.U64ToBytes(index)
+		params := convert.I32ToBytes(chunkIndex)
 
 		// Chunk data length encoding
 		params = append(params, convert.U32ToBytes(chunkSize)...)
 
 		// Chunk data encoding
-		params = append(params, chunks[index]...)
+		params = append(params, chunks[chunkIndex]...)
+
+		// Total cost is the cost for storage bytes plus a fixed cost for key creation
+		uploadCost := sendOperation.StorageKeyCreationCost + sendOperation.StorageCostPerByte*chunkSize
+
+		if chunkIndex == 0 {
+			// if chunkID == 0, we may need to add the cost of the key creation for the NB_CHUNKS key
```

Comment:
Why the may ? Isn't it mandatory?

Reply:
Its bad copy pasta from the UploadMissedChunks! removing the "may".
in the missed chunk upload context, the NB_CHUNKS will already exist.
But the value we provide here is for "maxCoins" so it it already exists, the coins will not be spent